File: algorithms/SearchAndSort/BubbleSort/bubbleSort.py
def BubbleSort(sortList):
    for nums in range(len(sortList) - 1, 0, -1):
        for i in range(nums):
            if sortList[i] > sortList[i+1]:
                temp = sortList[i]
                sortList[i] = sortList[i+1]
                sortList[i+1] = temp
    return sortList

myList = [1, 5, 3, 9, 10, 20, 11]
BubbleSort(myList)
print(myList)

def ShortBubbleSort(sortList):
    exchange = True
    nums = len(sortList) - 1
    while nums > 0 and exchange:
        exchange = False
        for i in range(nums):
            if sortList[i] > sortList[i+1]:
                exchange = True
                temp = sortList[i]
                sortList[i] = sortList[i+1]
                sortList[i+1] = temp
        nums = nums - 1

myList = [1, 3, 5, 10]
ShortBubbleSort(myList)
print(myList)

File: algorithms/SearchAndSort/BubbleSort/bubbleSort.js
function BubbleSort(sortList) {
    for(var i = sortList.length - 1; i >= 0; i--) {
        for (var j = 0; j < i; j++) {
            if(sortList[j] > sortList[j+1]) {
                temp = sortList[j]
                sortList[j] = sortList[j+1]
                sortList[j+1] = temp
            }
        }
    }
    return sortList
}

var myList = [1, 10, 2, 5, 9]
console.log(BubbleSort(myList))

function ShortBubbleSort(sortList) {
    var exchange = true
    var sortListNum = sortList.length - 1
    while (sortListNum > 0 && exchange) {
        exchange = false
        for (var i = 0; i < sortListNum; i++) {
            if(sortList[i] > sortList[i+1]) {
                exchange = true
                temp = sortList[i]
                sortList[i] = sortList[i+1]
                sortList[i+1] = temp
            }
        }
        sortListNum--
    }
    return sortList
}

var myList = [1, 10, 9, 2]
console.log(ShortBubbleSort(myList))